### PR TITLE
deposits: improve editor and process

### LINKED ISSUE #560: Remove default project entry in deposit process
This will solve this use case:
* a moderator checks a deposit without project. Now she/he has to remove the empty project entry to be able to continues his/her validation to the next step.

### LINKED ISSUE #569: Improve deposit editor
- The identifier sub-fields are grouped with a grey vertical bar (see screenshot). It would be nice to also group:
    - Title in other language: language, title
    - Contained in (host document): document, year, pages, publisher, editors
image

- Replace "editors" by "authors/editors", because in French they have the same translation:
image

- Remove the field validation on "editors"/"authors". We should be able to enter a corporate body (ex. "HEP BEJUNE") as author, without comma.
- Replace "Contained in (host document)" by "Contained in (journal, book, proceedings)"
- Remove the field "editors"/"authors" for document type "Journal article" (c_6501)
- `partOf`: `volume` and `number` should be validated and contain only integer.
- `partOf`: is it possible to remove the field from the deposit editor: at the document creation, `provisionActivity.startDate` and `parfOf.year` use both the document date of the deposit (only for 'coar:c_3248', 'coar:c_5794', 'coar:c_6670', 'coar:c_3e5a', 'coar:c_beb9', 'coar:c_6501', 'coar:c_998f', 'coar:c_dcae04bc')?
- rename "Subjects" into "Keywords" (also in singular)
- step 5, field "license", set as default value "License undefined"
    - in the license enum, change the two values
        - "Other OA / license undefined" > "License undefined"
        - "Not OA / Rights reserved" > "Rights reserved"
- step 5, add field "oa_status", optional, with label "Open access status"
- step 4 (projects)
    - add a date picker for the 2 dates
    - move the examples for the 2 dates ("Example: 2019-05-05") from the description (tooltip in the form) to the placeholder, in order to be compliant with our [guidelines](https://github.com/rero/developer-resources/blob/master/interface/layout-charter.md#4-form-design).

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Improves editor and process for deposits.
* Closes #569.
* Closes #560.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

**Review by pronguen — COMMENTED** [review]
Nice enhancements, thank you!

Two details:

* the validation should provide a message to the end user for field `partOf.volume`, `partOf.number` and `partOf.pages`. See screenshot.
image
* When adding a keyword, the form should automatically open the value field.
image

File: sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json [hunk 1]
```diff
@@ -560,25 +558,38 @@
...
-            "year": {
-              "title": "Year",
-              "type": "string",
-              "minLength": 1
-            },
             "volume": {
               "title": "Volume",
               "type": "string",
               "minLength": 1,
+              "pattern": "^[0-9]+$",
```
Comment (zannkukai) [inline sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json:565]: from my experience, we coud have many unsuspected values in these kin of fields : `January`, `extra vol.1`, ....
If you force a pattern here, users will try to add this info into another inappropriate field.

File: sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json [hunk 2]
```diff
@@ -560,25 +558,38 @@
...
+                    "patternMessage": "Please enter a valid number."
+                  }
+                }
               }
             },
             "number": {
               "title": "Number",
               "type": "string",
               "minLength": 1,
+              "pattern": "^[0-9]+$",
```
Comment (zannkukai) [inline sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json:582]: same here : `extra 2021`, `special number # 1`, ...

File: sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json:565]: As discussed with @pronguen, at this time we will keep this constraint

**Pull request merged**